### PR TITLE
Dont project refs

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

**Comment by jewzaam** [comment]
@bserdar If I remove the Translator change the unit tests all pass.  I would expect it to fail if there was something to fix.  Am I missing something?

**Comment by bserdar** [comment]
You are right. Looking...

File: crud/src/main/java/com/redhat/lightblue/crud/mongo/Translator.java [hunk 1]
```diff
@@ -380,21 +381,31 @@ public DBObject translate(EntityMetadata md, UpdateExpression expr)
...
-            }
+        if(cursor.next()) {
+            boolean done=false;
+            do {
+                Path field=cursor.getCurrentPath();
+                FieldTreeNode node=cursor.getCurrentNode();
+                LOGGER.debug("Checking if {} is included ({})",field,node);
+                if(node instanceof ResolvedReferenceField||
+                   node instanceof ReferenceField) {
+                    done=!cursor.nextSibling();
```
Comment (jewzaam) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/Translator.java]: Couldn't this stop too early?  What if the ref field is in the middle of the cursor?  What if there are multiple references in the query?

Comment (bserdar) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/Translator.java]: It shouldn't stop early. cursor.next moves depth first. When we see a
resolvedReference, it calls nextSibling, i.e. should skip the whole subtree
rooted at that reference, and continue depth first.

**Pull request merged**